### PR TITLE
gdbm: initial integration

### PR DESCRIPTION
**Pull request by graygnuorg**

### CODE REVIEW

**Comment by graygnuorg** [comment]
@googlebot I signed it!

**Review by inferno-chromium — CHANGES_REQUESTED** [review]
```
Changed files: projects/gdbm/project.yaml
Error in /home/runner/work/oss-fuzz/oss-fuzz/projects/gdbm/project.yaml: main_repo section is missing.
Failed.
Error: Process completed with exit code 1.
```

**Comment by DavidKorczynski** [comment]
@graygnuorg the `project.yaml` file alone won't enable fuzzing: are you also intending on including a `Dockerfile`, `build.sh` and fuzzers (preferably those should be in the gdbm repository) in this PR?

**Comment by inferno-chromium** [comment]
will let @AdamKorcz @DavidKorczynski review and merge/suggest changes.

File: projects/gdbm/build.sh [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+upgrade 
+downgrade 
+close 
+quit
+EOF
+
+# Create seed
+mkdir seed
+cd seed
+for format in standard numsync; do
```
Comment (DavidKorczynski) [inline projects/gdbm/build.sh:62]: The build script gets a bit bloated IMHO having the seeds like this - would it make sense to make them explicit and place them in the repository here? https://git.gnu.org.ua/gdbm.git/tree/fuzz?h=fuzz

**Review by DavidKorczynski — CHANGES_REQUESTED** [review]
Looks good but what about moving the seeds to the upstream repo? Following that then this looks good to me.

File: projects/gdbm/build.sh [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+$CC $CFLAGS -c -I.. -I../src -ogdbm_fuzzer.o gdbm_fuzzer.c
+$CXX $CFLAGS -ogdbm_fuzzer gdbm_fuzzer.o ../src/libgdbmapp.a ../src/.libs/libgdbm.a $LIB_FUZZING_ENGINE
+
+cp gdbm_fuzzer $OUT
+cp gdbm_fuzzer.rc $OUT
+
+# Create seed
+PATH=$SRC/gdbm/src:$PATH sh ./build_seed.sh -C seed
+zip -rj "$OUT/gdbm_fuzzer_seed_corpus.zip" seed/
+
```
Comment (DavidKorczynski) [inline projects/gdbm/build.sh:34]: Nit: 3 blank lines here

**Review by DavidKorczynski — APPROVED** [review]

**Review by DavidKorczynski — APPROVED** [review]

**Comment by DavidKorczynski** [comment]
@graygnuorg please remove the blanks lines highlighted above and I will merge this in.

**Review by DavidKorczynski — APPROVED** [review]
LGTM. Thanks @graygnuorg 

@inferno-chromium I think you will have to merge this one is as you requested changes (I can't merge because of that).

**Review by inferno-chromium — APPROVED** [review]

**Pull request merged**